### PR TITLE
[DatePicker] fix onChange on month change after initial day selection

### LINKED ISSUE #3340: Date Picker: date doesn't update the first time the month is changed
#### Environment

- __Package version(s)__: datetime 3.7.0 (latest)
- __Browser and OS versions__: Chrome 72.0.3626.81, Firefox 65.0

#### Steps to reproduce

1. Refresh whatever page you are working on.
1. Select a date. It doesn't matter if you already navigated between months/years.
1. Use the month/year menus or left/right arrows to go to a different month or year.
1. Note that the `onChange` event has not fired.
1. Now use month/year menus or left/right arrows again.
1. Note that the `onChange` event does fire, supplying the new selected date and setting `isUserChange` to `false`.

You can reproduce it with the example on https://blueprintjs.com/docs/#datetime/datepicker . Note how the blue display below the calendar does not change after step 3, but does after step 5.

#### Actual behavior

The `onChange` event only fires the second time a month/year navigation happens, not the first.

#### Expected behavior

`onChange` should fire both times.

#### Possible solution

Using the component in a controlled manner may be a workaround. It's significantly more complex and I want to ignore non-user-initiated date changes anyway, so I have no personal interest in going down that road.

Edit because it turns out controlled usage actually works better for my use case: controlled usage doesn't have this bug, only uncontrolled.

### PR DESCRIPTION
**Pull request by EndiM**

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @EndiM! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: packages/datetime/src/datePicker.tsx [hunk 1]
```diff
@@ -246,7 +246,7 @@ export class DatePicker extends AbstractPureComponent<IDatePickerProps, IDatePic
                 selectedDay: day.getDate(),
             });
         }
-        if (this.state.value == null || this.state.value.getMonth() !== day.getMonth()) {
+        if (this.state.value !== null && this.state.value.getMonth() !== day.getMonth()) {
```
Comment (giladgray) [inline packages/datetime/src/datePicker.tsx]: `!= null` checks for null or undefined. please change.

Comment (EndiM) [reply packages/datetime/src/datePicker.tsx]: Will change to `!=` and add test @giladgray

Comment (EndiM) [reply packages/datetime/src/datePicker.tsx]: Changed to `!=` operator and added one test.

**Pull request merged**